tiledb/api/c_api/buffer/buffer_api_internal.h

```diff
@@ -78,6 +78,10 @@ struct tiledb_buffer_handle_t
   [[nodiscard]] inline const tiledb::sm::Buffer& buffer() const {
     return buffer_;
   }
+
+  operator span<const char>() const& {
```

Comment:
I am not sure about our policy here, but I think it is preferred to not have implicit conversions in code, or at least try to minimize them. So perhaps we should use helper methods like: `buffer->as_span()` or even free standing function: `tiledb::as_span(buffer)` overloaded for each type that we use. This is of course just a suggestion, feel free to ignore.

Reply:
I personally prefer it because some other C++ types can implicitly convert to span, but might revisit this approach as work on the PR is being made.

Reply:
Implicit conversion operator removed from `tiledb_buffer_handle_t` and now exists on `SerializationBuffer`.